### PR TITLE
Use `NameOf` instead of explicit string literals.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Change to use VB.net's `NameOf` operator, instead of explicit strings.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb [hunk 1]
```diff
@@ -195,7 +195,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Throw New ArgumentException("Must resolve overloads on PropertySymbol or MethodSymbol", "TMember")
             End If
             If isProperties And Not typeArguments.IsEmpty Then
-                Throw New ArgumentException(VBResources.PropertiesCanNotHaveTypeArguments, "typeArguments")
+                Throw New ArgumentException(VBResources.PropertiesCanNotHaveTypeArguments, NameOf(members))
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb]: This is incorrect.

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 2]
```diff
@@ -750,7 +750,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                    bindingOption As SpeculativeBindingOption,
                                                    <Out> ByRef binder As Binder) As BoundNodeSummary
             If expression Is Nothing Then
-                Throw New ArgumentNullException("expression")
+                Throw New ArgumentNullException(NameOf(position))
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: incorrect

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 3]
```diff
@@ -833,7 +833,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Private Function GetSpeculativelyBoundAttributeSummary(position As Integer, attribute As AttributeSyntax, <Out> ByRef binder As Binder) As BoundNodeSummary
             If attribute Is Nothing Then
-                Throw New ArgumentNullException("attribute")
+                Throw New ArgumentNullException(NameOf(position))
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: incorrect

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 4]
```diff
@@ -2058,10 +2058,10 @@ _Default:
             CheckPosition(position)
 
             If symbol Is Nothing Then
-                Throw New ArgumentNullException("symbol")
+                Throw New ArgumentNullException(NameOf(position))
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: incorrect

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 5]
```diff
@@ -2311,10 +2311,10 @@ _Default:
         ''' type), use Compilation.ClassifyConversion.</remarks>
         Public Shadows Function ClassifyConversion(position As Integer, expression As ExpressionSyntax, destination As ITypeSymbol) As Conversion
             If destination Is Nothing Then
-                Throw New ArgumentNullException("destination")
+                Throw New ArgumentNullException(NameOf(position))
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: I'm going to stop reviewing here as it seems it is as much wrong as right.

**Comment by gafter** [comment]
Good idea, but bad execution. No thanks.

**Comment by AdamSpeight2008** [comment]
@gafter I've pushed an update correcting the error introduce by the tool I used. I've gone through each one and made sure that they are all correct.

**Comment by sharwell** [comment]
@gafter Considering how easy it is to write automated transforms using Roslyn, it's a pretty safe bet that if you see obvious errors in something otherwise simple like this you've found a bug in someone's tools. :+1:

**Comment by AdamSpeight2008** [comment]
@gafter @sharwell  I should have at least checked the diff. :mag:

**Comment by gafter** [comment]
Looks good to me. @agocke does this look good to you?

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/ObjectDisplay.vb [hunk 6]
```diff
@@ -1,5 +1,4 @@
-﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
```
Comment (agocke) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/ObjectDisplay.vb]: I think we need the copyright notice.

**Comment by agocke** [comment]
Barring the missing copyright notice, LGTM

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/ObjectDisplay.vb (hunk 6 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/ObjectDisplay.vb]: I can't remember changing that but will fix it.

**Comment by gafter** [comment]
Integrated in https://github.com/dotnet/roslyn/commit/8b65942b9d4921b21e1356a54530c366b4d6b170

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 7]
```diff
@@ -2058,10 +2058,10 @@ _Default:
             CheckPosition(position)
 
             If symbol Is Nothing Then
-                Throw New ArgumentNullException("symbol")
+                Throw New ArgumentNullException(NameOf(symbol))
             End If
 
-            Dim vbsymbol = symbol.EnsureVbSymbolOrNothing(Of symbol)("symbol")
+            Dim vbsymbol = symbol.EnsureVbSymbolOrNothing(Of symbol)(NameOf(position))
```
Comment (mikedn) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: @AdamSpeight2008 @gafter This appears to have been accidentally changed from symbol to position.

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 8]
```diff
@@ -3311,14 +3311,14 @@ _Default:
         End Function
 
         Protected NotOverridable Overrides Function AnalyzeDataFlowCore(firstStatement As SyntaxNode, lastStatement As SyntaxNode) As DataFlowAnalysis
-            Return Me.AnalyzeDataFlow(SafeCastArgument(Of StatementSyntax)(firstStatement, "firstStatement"),
-                                                SafeCastArgument(Of StatementSyntax)(lastStatement, "lastStatement"))
+            Return Me.AnalyzeDataFlow(SafeCastArgument(Of StatementSyntax)(firstStatement, NameOf(firstStatement)),
+                                                SafeCastArgument(Of StatementSyntax)(lastStatement, NameOf(firstStatement)))
```
Comment (mikedn) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: @AdamSpeight2008 @gafter This appears to have been accidentally changed from lastStatement to firstStatement.

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 9]
```diff
@@ -3337,12 +3337,12 @@ _Default:
         End Function
 
         Protected NotOverridable Overrides Function AnalyzeControlFlowCore(firstStatement As SyntaxNode, lastStatement As SyntaxNode) As ControlFlowAnalysis
-            Return Me.AnalyzeControlFlow(SafeCastArgument(Of StatementSyntax)(firstStatement, "firstStatement"),
-                                                   SafeCastArgument(Of StatementSyntax)(lastStatement, "lastStatement"))
+            Return Me.AnalyzeControlFlow(SafeCastArgument(Of StatementSyntax)(firstStatement, NameOf(firstStatement)),
+                                                   SafeCastArgument(Of StatementSyntax)(lastStatement, NameOf(firstStatement)))
```
Comment (mikedn) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb]: @AdamSpeight2008 @gafter This appears to have been accidentally changed from lastStatement to firstStatement.

File: src/Compilers/VisualBasic/Portable/Compilation/SyntaxTreeSemanticModel.vb [hunk 10]
```diff
@@ -1372,10 +1372,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Overrides Function ClassifyConversion(expression As ExpressionSyntax, destination As ITypeSymbol) As Conversion
             CheckSyntaxNode(expression)
             If destination Is Nothing Then
-                Throw New ArgumentNullException("destination")
+                Throw New ArgumentNullException(NameOf(destination))
             End If
 
-            Dim vbdestination = destination.EnsureVbSymbolOrNothing(Of TypeSymbol)("destination")
+            Dim vbdestination = destination.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(expression))
```
Comment (mikedn) [inline src/Compilers/VisualBasic/Portable/Compilation/SyntaxTreeSemanticModel.vb]: @AdamSpeight2008 @gafter This appears to have been accidentally changed from destination to expression.

**Comment by AdamSpeight2008** [comment]
@mikedn @gafter Reopen and I'll push the changes.

**Comment by gafter** [comment]
Integrated in https://github.com/dotnet/roslyn/commit/8b65942b9d4921b21e1356a54530c366b4d6b170

**Comment by gafter** [comment]
@mikedn Thanks for catching those! I'll fix them immediately.

**Comment by AdamSpeight2008** [comment]
@gafter See #2018 for the fixes

**Pull request closed without merging**